### DISCUSSION TITLE
Assets - Multisite Propagation Method

### DISCUSSION THREAD

**Discussion by MaddieB-MWCompany** [craftcms/cms#15175] [Ideas] [answered]
It would be helpful to be able to customize the propagation method for assets in multisite setups. Currently, when you upload assets to a volume they are accessible across all sites. Being able to control how they are shared between sites (if at all) in a similar way to how entries are controlled, would be a very useful feature.

**Comment by zizther** [comment]
Supporting this would be a great feature.

My scenario is I have a multi site setup and have several users with access to multi sites, I need to limit which volume a user can upload an asset to so assets don't cross over.

**Comment by brandonkelly** [comment, marked as answer]
This doesn’t really make sense for assets, since asset elements represent actual files in the volume – which is just a location on the disk, or a bucket/container in a cloud storage provider. I don’t think we should pretend that files don’t exist when they do.

If you want to limit assets per-site, you should make site-specific subfolders on the disk/bucket/container, and create separate volumes that point to each one. (As of Craft 5, you would only need to create a single filesystem that points to the root location, and then each volume can use that FS and define the site-specific subpath.) Then only grant access to those volumes based on who has access to which sites, and/or create Assets fields for each volume, and add site-based condition rules to those Assets fields within field layouts, so they only show up for the correct site.
